libs/server-sdk/src/client_impl.cpp

```diff
@@ -157,15 +158,50 @@ bool ClientImpl::Initialized() const {
     return IsInitializedSuccessfully(status_manager_.Status().State());
 }
 
-std::unordered_map<Client::FlagKey, Value> ClientImpl::AllFlagsState() const {
+AllFlagsState ClientImpl::AllFlagsState(Context const& context,
+                                        AllFlagsState::Options options) {
     std::unordered_map<Client::FlagKey, Value> result;
-    // TODO: implement all flags state (and update signature).
-    //    for (auto& [key, descriptor] : memory_store_.AllFlags()) {
-    //        if (descriptor->item) {
-    //            result.try_emplace(key, descriptor->item->Value());
-    //        }
-    //    }
-    return result;
+
+    if (config_.Offline()) {
+        LD_LOG(logger_, LogLevel::kWarn)
+            << "AllFlagsState() called, but client is in offline mode. "
+               "Returning empty state";
+        return {};
+    }
+
+    if (!Initialized()) {
```

Comment:
This flow doesn't look right taking persistence into account. 

If the flags are in persistence, then we would evaluate all flags and provide a warning that it was called before evaluation. 

If persistence wasn't populated, and we were not initialized, then I would expect this. The variation internal method has it.

Node equivalent
```
    if (!this.initialized()) {
      const storeInitialized = await this.featureStore.initialized();
      if (storeInitialized) {
        this.logger?.warn(
          'Called allFlagsState before client initialization; using last known' +
            ' values from data store',
        );
      } else {
        this.logger?.warn(
          'Called allFlagsState before client initialization. Data store not available; ' +
            'returning empty state',
        );
        valid = false;
      }
    }
```

Reply:
Variation internal has something similar.